src/circllhist.c

```diff
@@ -88,15 +88,23 @@ static const char __b64[] = {
   'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
   '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '+', '/', 0x00 };
 
+struct hist_flevel {
+  uint8_t l2;
+  uint8_t l1;
+};
 struct histogram {
   uint16_t allocd;
   uint16_t used;
+  uint32_t fast:1;
   struct {
     hist_bucket_t bucket;
     uint64_t count;
   } __attribute__((packed)) *bvs;
 };
-
+struct histogram_fast {
+  struct histogram internal;
+  uint16_t *faster[256];
```

Comment:
array of pointers.  two-level cache.  2k + 512b per populated level 2.  As opposed to a single slab of 128k